### PR TITLE
Add scalar reference reading for attributes

### PR DESCRIPTION
**Pull request by grahamrow**
Wasn't able to read a single reference object from an attribute (only arrays) — quick fix.

### CODE REVIEW

File: src/HDF5.jl [hunk 1]
```diff
@@ -1258,6 +1258,13 @@ function read{A<:FixedArray}(obj::DatasetOrAttribute, ::Type{Array{A}})
     ret
 end
 
+# Read a scalar reference objected from an attribute
+function read{T<:HDF5ReferenceObj}(obj::HDF5Attribute, ::Type{T})
+    refs = Array{T}(1)
```
Comment (musm) [inline src/HDF5.jl]: Would this not be better of using `Ref{T}()`?

**Comment by simonster** [comment]
It would be good to have a test for this, if it's easy to make one (i.e., if we can write a reference to an attribute).

File: src/HDF5.jl (hunk 1 above)
Comment (grahamrow) [inline src/HDF5.jl]: True, but more lifting would be required since `getindex` doesn't support Ref at the moment.

Comment (simonster) [inline src/HDF5.jl]: You can use `ref[]` to get the contents of a `Ref`.

**Comment by grahamrow** [comment]
I was working on a test, for which one requires the ability to write a single reference to an attribute. It seems like one must override `a_create` and `a_write` for `HDF5ReferenceObjs`, but I'm having trouble figuring out the correct dataspace for a reference object.

**Comment by grahamrow** [comment]
@simonster It seems like this is working now. Let me know if anything else needs tweaking.

**Comment by musm** [comment]
I rebased this for you @grahamrow

**Comment by musm** [comment]
Since a test has been added, planing on merging this Saturday

**Comment by grahamrow** [comment]
Sounds good, thanks for the comments.

**Comment by simonster** [comment]
I think #380 is a better way of handling this.

**Comment by musm** [comment]
close in favor of https://github.com/JuliaIO/HDF5.jl/pull/380 ?

**Pull request closed without merging**